### DISCUSSION TITLE
Using Django / ORM

### DISCUSSION THREAD

**Discussion by shivan-s** [mrdbourke/nutrify#15] [Ideas]
There was talk of using SQL to store the image metadata.

To forgo writing SQL, we can use an ORM (object relational mapper). This saves us having to write complicated SQL, as an ORM turns your python code into SQL.

Examples of an ORM are of course Django (https://www.djangoproject.com/start/). This saves us having to deal with creating tables, or changing rows with SQL thanks to django migrations (happy to explain this further). This can save us headaches in the future.

Another advantage is that we can use Django REST framework (https://www.django-rest-framework.org/tutorial/quickstart/) and create an API for others to access the metadata.

Let me know your thoughts.

**Comment by shivan-s** [comment]
@mrdbourke you do you bro, but friends shouldn't let other friends suffer also.

Please consider the above before diving deep into SQL. Note that django can link to postgresql and if you want to create users and authorisation and authentication then this would be the way to go.

All thr best.

**Comment by shivan-s** [comment]
At the moment, I'm writing a v2 with Django and postgresql as the database.

**Comment by mrdbourke** [comment]
Hey hey! Very nice thank you for the update.

I started on PostgreSQL today for metadata (just replicating the same as what's in Google Sheets).

User authentication can come later when user accounts are required - though if you know how to do this, that'd be great :D

**Comment by shivan-s** [comment]
@mrdbourke how is the project going?

Here is a the django version 2 of the application:

https://nutrify-app-v2.herokuapp.com/

Comes with:
- User auth
- Anon can upload
- Public can see images
- Users who upload images will be logged against them so you can gain 'points'

Source code is on my repo as a forked project of this: https://github.com/shivans93/nutrify

Please let me know what you think. Lot's of fun making this!

Shivan
